database/prefixdb/db.go

```diff
@@ -157,7 +199,16 @@ func (db *Database) Close() error {
 }
 
 func (db *Database) prefix(key []byte) []byte {
-	prefixedKey := make([]byte, len(db.dbPrefix)+len(key))
+	// Get a []byte from the pool
+	prefixedKey := db.bufferPool.Get().([]byte)
+	keyLen := len(db.dbPrefix) + len(key)
+	if cap(prefixedKey) >= keyLen {
+		// The [] byte we got from the pool is big enough to hold the prefixed key
+		prefixedKey = prefixedKey[:keyLen]
+	} else {
+		// The []byte from the pool wasn't big enough...allocate a new, bigger one
+		prefixedKey = make([]byte, keyLen)
```

Comment:
Should we `db.bufferPool.Put(prefixedKey)` before allocating the new array?

Reply:
Yeah, we can do that...the reason I didn't is that if the byte buffer is too small and we put it back in the pool, then it might get taken from the pool and put back repeatedly before finally getting garbage collected.

In practice, I think all the keys are going to be smaller than 512 bytes so it shouldn't be an issue.